include/mbgl/gl/location_component_layer_factory.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
```

Comment:
Since this layer type is included in the style spec, shouldn't this file live in `include/mbgl/layermanager/`?

Reply:
Same for implementation files.

Reply:
This blocks the generation of the platform binding because the path is hardcoded
https://github.com/mapbox/mapbox-gl-native/blob/488e5508959573b4b84a4f03afc7589503cd0af1/platform/android/src/style/layers/layer.hpp.ejs#L11-L12